lib/puppeteer-to-istanbul.js

```diff
@@ -32,8 +34,12 @@ class PuppeteerToIstanbul {
       let istanbulCoverage = script.toIstanbul()
       let keys = Object.keys(istanbulCoverage)
 
-      let jsonPart = {}
-      jsonPart[keys[0]] = istanbulCoverage[keys[0]]
+      if (jsonPart[keys[0]]) {
+        // Merge coverage records
+        mergeCoverageData(jsonPart[keys[0]].s, istanbulCoverage[keys[0]].s)
```

Comment:
@joe223

If `mergeCoverageData` is called here, the change will not be reflected in the `out.json` file, since the `jsonPart` has already been written to disk in a previous iteration of the loop.

I suggest a second loop, before this one, that will merge all data as necessary and will build a complete JSON. Then the current loop will go through all the keys to write them to disk.

Reply:
Change will be reflected in the `out.json` because of `jsonPart` is closure variable in top scope. But the coverage records will be repeated writes over and over again (just as `puppeteerToV8Info.length` times).

I have fixed this problem with latest commit and added test case.

Thanks for reminder.